app/src/main/java/com/moviefiend/torian/moviefiend/MainActivity.java

```diff
@@ -1,13 +1,26 @@
 package com.moviefiend.torian.moviefiend;
 
+import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 
-public class MainActivity extends AppCompatActivity {
+import com.moviefiend.torian.moviefiend.network.NowPlayingResponse;
+
+public class MainActivity extends AppCompatActivity implements MovieListFragment.Listener {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        MovieListFragment fragment = (MovieListFragment) getFragmentManager().findFragmentById(R.id.movie_list_fragment);
+        fragment.setListener(this);
```

Comment:
I get your point about the potential danger but:
1. Can a fragment have a destroyed activity? isn't the fragment destroyed first anyway?
2. If I can have a fragment with a destroyed activity as you say, how referencing to getActivity() is any better? it will also result in accessing an invalid pointer